.travis.yml

```diff
@@ -1,23 +1,29 @@
 # http://travis-ci.org/#!/jupyter/jupyterhub
 language: python
-sudo: false
 python:
-    - 3.7-dev
-    - 3.6
-    - 3.5
-    - 3.4
     - 3.3
+    - 3.4
+    - 3.5
+    - 3.6
+    - nightly
+
+# installing dependencies
 before_install:
     - npm install
     - npm install -g configurable-http-proxy
     - git clone --quiet --depth 1 https://github.com/minrk/travis-wheels travis-wheels
 install:
     - pip install --pre -f travis-wheels/wheelhouse -r dev-requirements.txt .
+
+# running tests
 script:
-    - travis_retry py.test --cov=jupyterhub jupyterhub/tests -v
+    - if [[ $TRAVIS_PYTHON_VERSION == '3.6' || $TRAVIS_PYTHON_VERSION == 'nightly' ]]; then pytest --maxfail=2 --cov=jupyterhub jupyterhub/tests; fi # for Python 3.6 and above
+    - if [[ $TRAVIS_PYTHON_VERSION == '3.5' || $TRAVIS_PYTHON_VERSION == '3.4' || $TRAVIS_PYTHON_VERSION == '3.3' ]]; then py.test --maxfail=2 --cov=jupyterhub jupyterhub/tests; fi # for < 3.5
```

Comment:
I pulled this from an older travis config on another project. It was to capture when py.test and pytest were being migrated to pytest command for small performance gains based on what Python version was in the Travis VM. Now that the migration is done in pytest, I think it would be better to have them as one. The performance gain was minimal anyway (one less pip install --upgrade pytest on the older pythons at the time).